### PR TITLE
PPS Pixel data unpacking

### PR DESCRIPTION
**Pull request by fabferro**
#### PR description:

This PR modifies the PPS pixel unpacking code in view of the changes for Run3. 
The changes concern 13 out of 64 bits of the data word that encode the pixel coordinates: the "double column-pixel id" convention is changed into a "column-row" one, due to the use of a slightly different readout chip (PROC600 instead of PsiDIG46).
The back compatibility with Run2 data is ensured by means of era modifiers: Run3 convention becomes the default one and Run2 one is triggered by means of ctpps_201* era modifiers.
The part of the code packing the simulated data is also changed accordingly.

#### PR validation:

The detectors are not yet installed. They will be installed beginning of November 21. Private validation on lab data, Run2 data and full simulation was done.  

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

No backport expected.

### CODE REVIEW

**Comment by fabferro** [comment]
@jan-kaspar @mundim @clemencia @AndreaBellora FYI

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by jpata** [comment]
Could a unit test / test config be useful here?

**Comment by jpata** [comment]
Kind ping @fabferro. Just to clarify, I think this looks fine from the reco point of view, but typically we try to ask for the possibility (if it's not unreasonably complicated or useless) to have a test for new functionality, to be able to (at least in principle) run the code we sign. I think in this case a small test config on some testbeam data file could be helpful. Let me know what you think!

**Comment by jpata** [comment]
Could something be done in analogy to PR#34759, https://github.com/ChrisMisan/cmssw/blob/465b1657fb94e540dd43becf6ccb9073654b2e42/RecoPPS/Local/test/totemTiming_digiConverter_reco_cfg.py?

The test file could then perhaps be located in `/eos/cms/store/group/dpg_ctpps/`.

File: EventFilter/CTPPSRawToDigi/python/ctppsRawToDigi_cff.py [hunk 1]
```diff
@@ -113,6 +113,11 @@
 from EventFilter.CTPPSRawToDigi.ctppsPixelDigis_cfi import ctppsPixelDigis
 ctppsPixelDigis.inputLabel = cms.InputTag("rawDataCollector")
 
+from Configuration.Eras.Modifier_ctpps_2016_cff import ctpps_2016
+from Configuration.Eras.Modifier_ctpps_2017_cff import ctpps_2017
+from Configuration.Eras.Modifier_ctpps_2018_cff import ctpps_2018
+(ctpps_2016 | ctpps_2017 | ctpps_2018).toModify(ctppsPixelDigis, isRun3 = cms.bool(False) )
```
Comment (jpata) [inline EventFilter/CTPPSRawToDigi/python/ctppsRawToDigi_cff.py:119]: Slava pointed to me that the type does not need (should not) be specified here, sorry for missing it earlier.
```suggestion
(ctpps_2016 | ctpps_2017 | ctpps_2018).toModify(ctppsPixelDigis, isRun3 = False )
```

Comment (fabferro) [reply EventFilter/CTPPSRawToDigi/python/ctppsRawToDigi_cff.py:119]: I missed it too when I re-read the code before submission. Slava never misses! ;-)
Thanks.
BTW, it works fine as well...

**Comment by fabferro** [comment]
The test config just added reads lab data from  /eos/cms/store/group/dpg_ctpps/comm_ctpps/PixelAlive_562_RAW.root and produces a root file with DIGIs applying a dummy DAQ map (/eos/cms/store/group/dpg_ctpps/comm_ctpps/CTPPSPixel_DAQMapping_AnalysisMask.db). The output, if limited to a few events only, can be manually compared with a binary-to-txt-dump file /eos/cms/store/group/dpg_ctpps/comm_ctpps/Run562_dumpBinary.txt.

**Comment by jpata** [comment]
@cmsbuild please test

**Comment by jpata** [comment]
+reconstruction
- for 20537f2
- no relevant reco changes expected or observed (1325.81 is from https://github.com/cms-sw/cmssw/issues/32628)
- a test configuration is provided and runs
- slides in https://indico.cern.ch/event/1071803/contributions/4507619/attachments/2301676/3915262/PPS_Reco_030921.pdf

**Comment by perrotta** [comment]
+1

**Pull request merged**